backend/src/utils/URLHandler.ts

```diff
@@ -57,6 +57,9 @@ export class URLHandler {
   githubURL: string | null = null; // the GitHub repository URL
   baseAPI: string | null = null;  // the base API URL
   constructor(url: string) {
+    if(url.startsWith("git+")) {
+      url = url.replace("git+", "");
+    }
```

Comment:
I think this logic is a little flawed. I think we should not add anything in the constructor. What you added is not thorough enough for all forms of git urls like "git@" and "git:" , but I also think it would make more sense code wise to change the checks in setRepoUrl function.

In setRepoUrl, we need to change the check: this.url.startsWith('https://github.com/')  to this.url.includes('github.com/') and then use URLHandler.convertGithubURLToHttps like you did below